lua/core/which-key.lua

```diff
@@ -236,6 +249,19 @@ M.config = function()
   }
 end
 
+-- Clean up mappings for inactive core plugins
+local function clean_up_inactive_mappings()
+  -- DAP
+  if not lvim.builtin.dap.active and lvim.builtin.which_key.mappings.d.name == "Debug" then
+    lvim.builtin.which_key.mappings.d = nil
+  end
```

Comment:
This will backfire sadly, because you're restricting what `d` and `e` can used for. You might end up removing more than necessary, or even removing the user's own configuration.

A better approach is either:
a) fetch the config from each of those modules, but only assign it if the plugin is active.
b) keep them in this file but don't add them unless the plugin is installed.

Reply:
@kylo252 

Yeah, I did try that first. But it seems that the problem is loading the keymaps in the plugin's `config()` does not work universally. 

Here is an example: if the user remaps the `<leader>d` to be a set of mappings dealing with say docker, then loading `d` selectively if `lvim.builtin.dap.active == true` might not be the correct outcome.

Maybe going forward, it might make sense to have some kind of metatable hacks so that each of the mappings gets a lambda to check when a mapping should be enabled or not. Then when they are loaded, we create a wrapper around the `register()` to check and register.